### DISCUSSION TITLE
[Node-Red] How to set Alarm-Screen when alarm is active

### DISCUSSION THREAD

**Discussion by davipet1** [pixelit-project/PixelIt#162] [Q&A] [answered]
Hi,

I am quiet a noob with Node Red and I wonder how to get an alarm-screen. I have a numeric fault output of my boiler in iobroker.
It is 0 when its all fine. When it is not 0 I want to have an alarm-screen as long as the number is != 0.

In my tests I only get the alarm-screen for the defined duration time. And when the alarm is set to 0 it also triggers the alarm-screen for the defined duration time.
Can someone please help me understand how that is meant to work? :)

Thanks!

David

P.S.: Gerne auch auf deutsch ;)

**Comment by davipet1** [comment]
OK, dann versuche ich das mal auf deutsch besser auszudrücken.

Ich bekomme also von meiner Heizung einen Fehlercode im iobroker.
Solange dieser Fehlercode 0 ist, ist alles OK und es soll kein Screen für die Heitung angezeigt werden.
Wenn der Fehlercode aber ungleich 0 ist möchte ich einen alarmscreen anzeigen. Und zwar so lange wie der Fehlercode ungleich 0 ist. 
Ich bekomme es nur so hin, dass bei Änderung des Fehlercodes der Alarm für z.B. 10sec (Duration time) angezeigt wird. Ebenso auch, wenn sich der Fehlercode auf 0 ändert - in dem Fall soll kein Alarm angezeigt werden.

Also bei Fehlercode ungleich 0 soll nur der Alarm angezeigt werden. Nichts anders, bis der Fehlercode wieder 0 ist.

Hast du eine Idee wie ich das abbilden kann?

Danke!
Gruß
David

**Comment by hamster65** [comment, marked as answer]
The alarm screen is shown for the set duration whenever you trigger it. It's like a pop up message that dissapears after some seconds.
If you want it to show as long as the fault condition is active, you have to re-trigger it every couple of seconds.

This can be done with the trigger node of node-red. It can resend messages every x seconds. 

Place another node behind it that filters out alarm values of zero to prevent the alert screen from showing up when there is no alert. I'd simply use a function node for this but there might be a node that does this.

**Reply by o0shojo0o** [reply]
Perfect answer :)

**Comment by davipet1** [comment]
Hi!
Yes - perfect! Thanks a lot.
I use the trigger node to resend the msg periodically. Behind the trigger node I tried to use the filter, but it filters only value changes not concrete values. So I use a function which only returns the msg if the payload isn´t 0.
Thanks for the help. I think i slowly get behind it ;)

Greetings

David
